tests/test_http_downloader.py

```diff
@@ -0,0 +1,61 @@
+import tempfile
+from unittest.mock import patch
+
+import pytest
+
+from datasetinsights.io.downloader.http_downloader import HTTPDownloader
+from datasetinsights.io.exceptions import ChecksumError
+
+
+def test_read_checksum_from_txt():
+    with tempfile.NamedTemporaryFile(mode="w+") as tmp:
+        tmp.write("123456")
+        tmp.read()
+        assert HTTPDownloader.get_checksum_from_file(tmp.name) == 123456
+
+
+@patch("datasetinsights.io.downloader.http_downloader.download_file")
+@patch.object(HTTPDownloader, "read_checksum_from_txt")
+def test_get_checksum_from_file(
```

Comment:
Unit test should follow the pattern. 
arrange 
act 
assert 

This will make the tests easier to understand. This unit test does multiple things and should instead be refactored into multiple tests.